git_repo/services/service.py

```diff
@@ -139,11 +139,13 @@ def __init__(self, r=None, c=None):
                 self.name = name
         # if not in the configuration file, retrieve the private key from the
         # environment (useful for travis configuration), otherwise, make it None.
-        # using "privatekey" or "private_token" in configuration file to avoid
+        # using "token" > "private_token" > "privatekey" in configuration file to avoid
         # confusion with the SSH keys (yes that happened).
+        # NB: `git config` doesn't parse underscores in option names, token.
         self._privatekey = os.environ.get('PRIVATE_KEY_{}'.format(self.name.upper()),
+                                          c.get('token',
                                           c.get('private_token',
-                                                c.get('privatekey', None)))
+                                          c.get('privatekey', None)))
```

Comment:
couple of things here:

* the indentation was made on purpose to show that `c.get('privatekey', None)` is within parenthesis of `c.get('token')` as a reminder to close all parenthesis!
* you forgot to close the third parenthesis as you've added `c.get('token', …)`:

```
os.environ.get('PRIVATE_KEY_{}', c.get('token', c.get('private_token', c.get('privatekey', None))))
                                                                                                  ^ missing parenthesis!
```

there, I'm glad to have implemented unit tests ☺